### ISSUE TITLE
Postman loses all history/collections and becomes unusable

### ISSUE THREAD

**Issue by tbrentlong** [a85/POSTMan-Chrome-Extension#1752]
This is the second time this has happened; the app crashes and completely disappears, and when I relaunch it, there are no history entries or collections, and the left hand pane obscures the main working area. I've had to reinstall to make it usable again.  Is anyone aware of this issue and the cause/resolution?

**Comment by jsteenGG** [comment]
I am also having the same issue.  All of my collections are gone!  HELP!

**Comment by ruiwen** [comment]
Same here! 

Chrome Dev 50.0.2657.0, Mac OS X 10.9.5

**Comment by ruiwen** [comment]
@tbrentlong Silly question, but are your collections and history present after (presumably uninstalling) and reinstalling? Or did you lose them all?

**Comment by sobtiankit** [comment]
We are investigating why the Chrome app is crashing. @ruiwen has it occurred earlier as well or since a recent app or Chrome update?

**Comment by ruiwen** [comment]
@sobtiankit Just happened to me with the latest update to Chrome Dev 50.0.2657.0

**Comment by ruiwen** [comment]
If it's any help @sobtiankit , the Dev Tools window for the app shows a couple of errors regarding localStorage

![screen shot 2016-02-24 at 12 53 34](https://cloud.githubusercontent.com/assets/307558/13275990/bced98fa-daf5-11e5-8f7e-1b9df5e16de7.png)

**Comment by sobtiankit** [comment]
Do you happen to have Sync enabled? Data should be recoverable by signing in. I'm going to get the crash with Chrome Dev investigated and get back to you shortly.

Thank you @ruiwen, the screenshot really helps. It seems a backward incompatible API change has been introduced in chrome apps.

If it's possible can you verify one more thing? This will help with data recovery. Can you share a screenshot of your indexedDb, like here: 

<img width="1676" alt="screen shot 2016-02-24 at 10 28 13 am" src="https://cloud.githubusercontent.com/assets/547061/13276240/50ca3b40-dae3-11e5-9ed5-9205c566fc4a.png">

**Comment by ruiwen** [comment]
@sobtiankit there you go

Sadly, I don't use Sync =(

Though it's a little bit of a relief to see that the data appears to be there at least

![screen_shot_2016-02-24_at_13_20_51](https://cloud.githubusercontent.com/assets/307558/13276379/95d810c0-daf9-11e5-8afd-00d8a1a8e0c8.png)

**Comment by sobtiankit** [comment]
Okay this is good. Which means, data is in the app. We'll push out an app today that uses the new chrome API for localStorage. Will update when that's done CC: @abhijitkane @numaanashraf

**Comment by ruiwen** [comment]
Sweet. Thanks so much @sobtiankit and team

Just wanna say, thanks for a great tool that's become indispensable at work everyday =)

**Comment by abhijitkane** [comment]
@ruiwen @jsteenGG 
We've released a fix for this in v3.2.19.
You can upgrade by heading to chrome://extensions, enabling 'Developer Mode', and click 'Update extensions now'.

**Comment by tbrentlong** [comment]
Thanks for looking at this. My first occurrence of it was probably a month ago, and none of my data was there after re-install.  I tried updating extensions, but it doesn't show me 3.2.19; it's still on 3.2.18.

**Comment by tbrentlong** [comment]
Looks like there are a few errors after trying to update.
![postman](https://cloud.githubusercontent.com/assets/11465123/13295495/0397ba0a-dade-11e5-882c-a2da901ec1b4.jpg)

**Comment by abhijitkane** [comment]
Does the page at chrome://extensions show 3.2.19 for Postman?

**Comment by jagrelo** [comment]
I was hitting the same problem and the update solved the issue. Thanks a lot for the quick fix!

**Comment by tbrentlong** [comment]
Mine is only showing 3.2.18 as available.  I'm using Version 50.0.2658.0 canary (64-bit)

**Comment by ruiwen** [comment]
@abhijitkane Thanks for the quick update! Have updated my app, and I've managed to get all my collections and data back!

**Comment by abhijitkane** [comment]
@tbrentlong A reinstall will always clear all your data. Is the page at https://chrome.google.com/webstore/detail/postman-rest-client-packa/fhbjgbiflinjbdggehcddcbncdddomop?hl=en showing 3.2.19 for you? If yes, can you try repeating these steps? (It's a 5.5MB download)

```
You can upgrade by heading to chrome://extensions, enabling 'Developer Mode', and click 'Update extensions now'.
```

**Comment by tbrentlong** [comment]
That page does show 3.2.19, but the chrome://extensions page shows 3.2.18.  I must have something out of joint.  I'll just go ahead and uninstall/reinstall and hope for the best in the future.  Thanks for tackling the issue so quickly!  Great tool.

**Comment by ruiwen** [comment]
Hi guys, 

Don't mean to be a nag, but I've lost all my collections and history again. And this time round, the IndexedDB section is empty. Any thoughts?

(Should I open a new issue?)

@abhijitkane 

Edit: This is on Chrome Version 51.0.2687.0 dev (64-bit), with Postman 4.1.2

![screen shot 2016-03-28 at 14 38 40](https://cloud.githubusercontent.com/assets/307558/14072432/fc89c288-f4f2-11e5-85dd-8f3941cc348e.png)

**Comment by abhijitkane** [comment]
@ruiwen Did you just open your app and see your data missing? Can you send the zipped contents of your indexedDB (head to chrome://indexeddb-internals/, search for fhbjgbiflinjbdggehcddcbncdddomop) to help@getpostman.com?

**Comment by ruiwen** [comment]
@abhijitkane Yup that's correct. Opened it this morning and it was all empty. (Chrome may have updated in the meantime)

There look like there are multiple hits for that search though.

There are a few access keys saved in my collections/history that I'd rather not have released I'm afraid. Could you guide me through instead?

**Comment by abhijitkane** [comment]
You could zip the contents, reinstall Postman, and unzip the backup in the same location.

**Comment by ruiwen** [comment]
@abhijitkane No luck I'm afraid. I tried copying those files to another Chrome profile where I had Postman installed, but the saved collections didn't show up. Tried copying the entire extension folder over either, but that didn't work too.

Is the uninstall/reinstall step critical here?

**Comment by ruiwen** [comment]
@abhijitkane Any further thoughts on this though? I've not been able to get my collections back at all

**Comment by abhijitkane** [comment]
@ruiwen Sorry, there's nothing we can do in case of IndexedDB corruption. If you didn't have Sync enabled, there's no way that we know of to recover your data.

**Comment by ruiwen** [comment]
@abhijitkane Is there a way to avoid having the IndexDB getting corrupted next time though? What are the potential causes of corruption?

**Comment by abhijitkane** [comment]
We're investigating possible causes, but documentation around this isn't comprehensive.

You can 1. Enable Sync, which will keep all your data in Sync with your account, or 2. Take regular backups (Settings > Data > Export) to your filesystem.

**Comment by ruiwen** [comment]
@abhijitkane Data loss happened again today. If it's any help to you guys, it happened after my Mac hung and required a reboot. I lost all collections and history again, despite having it been saved previously. Though, it seems like it might be happening in conjunction with Chrome upgrades on the dev channel.

**Comment by kurtkanaskie** [comment]
I had Sync enabled and lost a couple days of work since I rebooted. Any advice?

**Comment by abhijitkane** [comment]
@kurtkanaskie Did you reboot your app or logout of Postman? You can view the server-side state of your data here - https://app.getpostman.com/dashboard/collections/private?expand=1
If this isn't accurate, you can mail us at help@getpostman.com

**Comment by kurtkanaskie** [comment]
Its strange, I have 2 personas, "personal" and "work" that I manage in Google. I launch Postman from Google Apps, signed in as my "personal" persona and I see SYNCING with a blue background and I notice things are missing from the past few days. At one point, after I saw things missing, I launched and postman showed me all my stuff and IN SYNC with an orange background. I quickly downloaded everything not knowing how I got here. But none the less, I got my stuff back. I created another account under my "work" persona and imported. All is good, although I can't explain it. BTW the server-side shows out of date collections, associated with my "personal" persona.

Thanks for the tip on server-side.
